packages/datetime/src/dateTimePicker.tsx

```diff
@@ -96,14 +96,16 @@ export class DateTimePicker extends AbstractComponent<IDateTimePickerProps, IDat
     }
 
     public componentWillReceiveProps(nextProps: IDatePickerProps) {
-        if (nextProps.value != null) {
-            this.setState({
-                dateValue: nextProps.value,
-                timeValue: nextProps.value,
-            });
-        } else {
-            // clear only the date to remove the selected-date style in the calendar
-            this.setState({ dateValue: null });
+        if (this.props.value !== nextProps.value) {
```

Comment:
nit: would prefer this structure to avoid nesting everything so much:

```tsx
if (this.props.value === nextProps.value) {
    return;
} else if (nextProps.value != null) {
    // ....
} else {
    // ...
}
```